go_api/interfaces/admin_controller.go

```diff
@@ -0,0 +1,91 @@
+// AdminのControllerを定義する
+package controller
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/NUTFes/lottery/go_api/domain"
+	"github.com/NUTFes/lottery/go_api/usecase"
+	"github.com/labstack/echo/v4"
+)
+
+type adminController struct {
+	adminUsecase usecase.AdminUsecase
+}
+
+type AdminController interface {
+	IndexAdmin(c echo.Context) error
+	ShowAdmin(c echo.Context) error
+	CreateAdmin(c echo.Context) error
+	UpdateAdmin(c echo.Context) error
+	DeleteAdmin(c echo.Context) error
+}
+
+func NewAdminController(uu usecase.AdminUsecase) AdminController {
+	return &adminController{adminUsecase: uu}
+}
+
+// 全アドミンユーザーの取得
+func (u *adminController) IndexAdmin(c echo.Context) error {
+	admins, err := u.adminUsecase.FindAllAdmin()
+	if err != nil {
+		return err
+	}
+	return c.JSON(http.StatusOK, admins)
+}
+
+// idを指定してアドミンユーザーを取得
+func (u *adminController) ShowAdmin(c echo.Context) error {
+	id, _ := strconv.Atoi(c.Param("id"))
+	admin, err := u.adminUsecase.FindAdmin(id)
+	if err != nil {
+		return err
+	}
+	return c.JSON(http.StatusOK, admin)
+}
+
+// アドミンユーザーの作成
+func (u *adminController) CreateAdmin(c echo.Context) error {
+	name := c.QueryParam("name")
+	email := c.QueryParam("email")
+
+	admin := &domain.Admin{
+		Name:      name,
+		Email:     email,
+		CreatedAT: time.Now(),
+		UpdatedAT: time.Now(),
+	}
+	if err := u.adminUsecase.CreateAdmin(admin); err != nil {
+		return err
+	}
+	return c.JSON(http.StatusOK, admin)
+}
```

Comment:
```suggestion
func (a *adminController) CreateAdmin(c echo.Context) error {
	name := c.QueryParam("name")
	email := c.QueryParam("email")
	password := c.QueryParam("password")

	admin := &domain.Admin{
		Name:      name,
		Email:     email,
		Password: password,
		CreatedAT: time.Now(),
		UpdatedAT: time.Now(),
	}
	if err := a.adminUsecase.CreateAdmin(admin); err != nil {
		return err
	}
	return c.JSON(http.StatusOK, admin)
}
```
クエリパラメータに`password`がないので、passwordの値が入らず作成されます。